site/docs/advanced/flood.md

```diff
@@ -5,29 +5,119 @@ next: ./transformers.md
-This plugin is just as simple to set up but does a much better job at flood control.
-There isn't really any good reason to use [auto-retry](../plugins/auto-retry.md) over the [throttler plugin](../plugins/transformer-throttler.md).
-In some cases it may make sense to use both.
+Deal with it.
+
+We have some good ideas about how many requests you can perform, but the exact numbers are unknown.
+(If someone tells you the actual limits, they are not well-informed.)
+The limits are not simply hard thresholds that you can find out by experimenting with the Bot API.
+Rather, they are flexible constraints that change based on your bot's exact request payloads, the number of users, and other factors, not all of which are known.
+
+Here are a few misconceptions and false assumptions about rate limits.
+
+- My bot is too new to receive flood wait errors.
+- My bot does not get enough traffic to receive flood wait errors.
+- This feature of my bot is not used enough to receive flood wait errors.
+- My bot leaves enough time between API calls so it will not receive flood wait errors.
+- This particular method call cannot receive flood wait errors.
+- `getMe` cannot receive flood wait errors.
+- `getUpdates` cannot receive flood wait errors.
+
+All of these are wrong.
+
+Let's get to the things we _do_ know.
+
+## Safe Assumptions About Rate Limits
+
+From the [Bot FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this), we know a few limits that cannot be exceeded, ever.
+
+1. _"When sending messages inside a particular chat, avoid sending more than one message per second. We may allow short bursts that go over this limit, but eventually you'll begin receiving 429 errors."_
+
+   This one should be pretty clear. The auto-retry plugin handles this for you.
+
+2. _"If you're sending bulk notifications to multiple users, the API will not allow more than 30 messages per second or so. Consider spreading out notifications over large intervals of 8—12 hours for best results."_
+
+   **This only applies to bulk notifications,** i.e. if you proactively message many users.
+   If you are just responding to messages from users, then it is no problem to send 1,000 or more messages per second.
+
+   When the Bot FAQ says that you should _"consider spreading out notifications over large intervals"_, it this does not mean that you should add any artificial delays.
+   Instead, the main takeaway here is that sending bulk notifications is a process that will take many hours.
+   You cannot expect to message all users instantly at the same time.
+
+3. _"Also note that your bot will not be able to send more than 20 messages per minute to the same group."_
+
+   Again, pretty clear. Completely unrelated to bulk notifications or how many messages are sent in the group. And yet again, the auto-retry plugin will take care of this for you.
```

Comment:
Why are there three sentences on one line?
```suggestion
   Again, pretty clear.
   Completely unrelated to bulk notifications or how many messages are sent in the group.
   And yet again, the auto-retry plugin will take care of this for you.
```

Reply:
It's cause I'm dumb